Testing/functional/tests/cypress/integration/ui/services/menu.js

```diff
@@ -0,0 +1,68 @@
+const { AuthMethod } = require("../../../support/constants");
+
+function toggleServices(enabled = true) {
+    cy.configureSettings({
+        services: {
+            enabled: enabled,
+        },
+    });
+}
+
+function login(isMobile) {
+    if (isMobile) {
+        cy.viewport("iphone-6"); // Set viewport to 375px x 667px
+    }
+    cy.login(
+        Cypress.env("keycloak.username"),
+        Cypress.env("keycloak.password"),
+        AuthMethod.KeyCloak
+    );
+    cy.checkTimelineHasLoaded();
+}
+
+describe("Menu System when services is enabled", () => {
+    beforeEach(() => {
+        toggleServices(true);
+    });
+    it("Validate Toggle Sidebar", () => {
+        login(false);
+        cy.get("[data-testid=servicesLabel]")
+            .should("be.visible")
+            .should("have.text", "Services");
+        cy.get("[data-testid=sidebarToggle]").click();
+        cy.get("[data-testid=servicesLabel]").should("not.be.visible");
+        cy.get("[data-testid=sidebarToggle]").click();
+        cy.get("[data-testid=servicesLabel]").should("be.visible");
+    });
+
+    it("Side bar contains services nav link", () => {
+        login(false);
+        cy.get("[data-testid=menu-btn-Services-link]").should(
+            "have.attr",
+            "href",
+            "/services"
+        );
+        cy.get("[data-testid=sidebarToggle]").should("be.visible");
+    });
+
+    it("Side bar expands on login for desktop", () => {
+        login(true);
+        cy.get("[data-testid=servicesLabel]").should("not.be.visible");
+    });
+});
+
+describe("Menu system when services is disabled", () => {
+    beforeEach(() => {
+        toggleServices(false);
+    });
+    it("Side bar does not contain services nav link when services is disabled", () => {
+        cy.configureSettings({
+            services: {
+                enabled: false,
+            },
+        });
+        login(false);
+        cy.get("[data-testid=menu-btn-Services-link]").should("not.be.visible");
```

Comment:
The "not.exist" assertion should be used when the element should not be present in the DOM.

However, it's also important to note that assertions for non-existence near the start of tests will happily return true and march onward if they execute before everything is loaded.  That can be prevented by first checking that a reliable element is visible.  Cypress will wait for that element to appear (or for the timeout value to elapse).

Reply:
I've shuffled a few elements around to check for more reliable test structures.

The disabling of features doesn't remove the element from the DOM but set opacity to 0. So I need to stick with `not.be.visible`

I appreciate the insight in the loading caveats though haven't done testing like this in a long time so this will come in handy.

Reply:
Good catch; I didn't notice it was a `v-show` rather than a `v-if`.